src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,1148 @@
+        final int type = ast.getType();
+        if (type == TokenTypes.DOT) {
+            visitDotToken(ast, variables);
+        }
+        else if (type == TokenTypes.VARIABLE_DEF) {
+            visitVariableDefToken(ast);
+        }
+        else if (type == TokenTypes.IDENT) {
+            visitIdentToken(ast, variables);
+        }
+        else if (TokenUtil.isOfType(ast, CLASS_AND_INTERFACE)) {
+            visitClassOrInterfaceToken(ast);
+        }
+        else if (type == TokenTypes.PACKAGE_DEF) {
+            packageName = extractQualifiedName(ast.getFirstChild().getNextSibling());
+        }
+        else if (type == TokenTypes.LITERAL_NEW
+                && isInsideLocalAnonInnerClass(ast)) {
+            visitLocalAnonInnerClass(ast);
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST ast) {
+        if (TokenUtil.isOfType(ast, SCOPES)) {
+            logViolations(ast, variables);
+        }
+        else if (ast.getType() == TokenTypes.COMPILATION_UNIT) {
+            leaveCompilationUnit();
+        }
+        else if (ast.equals(blockContainingLocalAnonInnerClass)) {
+            blockContainingLocalAnonInnerClass = null;
+        }
+        else if (isNonLocalClassOrInterface(ast)) {
+            depth--;
+            classes.pop();
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#DOT}.
+     *
+     * @param dotAst dotAst
+     * @param variablesStack stack of all the relevant variables in the scope
+     */
+    private void visitDotToken(DetailAST dotAst, Deque<VariableDesc> variablesStack) {
+        if (blockContainingLocalAnonInnerClass == null
+                && dotAst.getParent().getType() != TokenTypes.LITERAL_NEW
+                && shouldCheckIdentTokenNestedUnderDot(dotAst)) {
+            checkIdentifierAst(dotAst.findFirstToken(TokenTypes.IDENT), variablesStack);
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#VARIABLE_DEF}.
+     *
+     * @param varDefAst varDefAst
+     */
+    private void visitVariableDefToken(DetailAST varDefAst) {
+        if (blockContainingLocalAnonInnerClass == null) {
+            addLocalVariables(varDefAst, variables);
+            addInstanceVarOrClassVar(varDefAst);
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#IDENT}.
+     *
+     * @param identAst identAst
+     * @param variablesStack stack of all the relevant variables in the scope
+     */
+    private void visitIdentToken(DetailAST identAst, Deque<VariableDesc> variablesStack) {
+        if (blockContainingLocalAnonInnerClass == null) {
+            final DetailAST parentAst = identAst.getParent();
+            if (!TokenUtil.isOfType(parentAst, UNACCEPTABLE_PARENT_OF_IDENT)
+                    && shouldCheckIdentWithMethodRefParent(identAst)) {
+                checkIdentifierAst(identAst, variablesStack);
+            }
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#CLASS_DEF}.
```

Comment:
Description and name of method aren't consistent.

Reply:
Done, explicitly mentioned types to be clear.

Reply:
Resolved.